code/modules/space transition/space_transition.dm

```diff
@@ -119,70 +119,41 @@ var/list/z_levels_list = list()
 	for(var/A in z_levels_list)
 		grid[A] = z_levels_list[A]
 
-	for(var/turf/space/S in world) //Define the transistions of the z levels
-		if(S.x <= TRANSITIONEDGE)
-			D = grid["[S.z]"]
-			if(!D)
-				CRASH("[S.z] position has no space level datum")
-			if(!(D.neigbours.len))
-				continue
-			if(D.neigbours[Z_LEVEL_WEST] && D.neigbours[Z_LEVEL_WEST] != D)
-				D = D.neigbours[Z_LEVEL_WEST]
-				S.destination_z = D.z_value
+	//Lists below are pre-calculated values arranged in the list in such a way to be easily accessable in the loop by the counter
+	//Its either this or madness with lotsa math
```

Comment:
This second comment should really say that it creates a border transition edge wide from the edges to search through rather than search all tiles one by one. Comment as is does not explain anything.